Remove the source maps wizard for react-native projects

smeubank opened the pull request:
### Description

The source maps wizard is designed for JavaScript projects. it cannot help developers working with react-native
- extends the logic to detect if it is a react-native project and remove the wizard on that condition.
- update copy in the modal for react-native projects for different states where we could not find the proper files with debugIds or debugIds were not properly injected
- added a generated link to the project's uploaded source maps, based on the given debugId
  - source maps page empty state to handle telling developers what they should to remedy that
  - configre their react-native project and importantly run scripts for iOS and Android builds

Updated the Source Maps Uploaded page:
- react-native project empty state directs users to trigger proper builds to upload source maps, and link to docs
- react-native project empty search state directs users to check for uploads of iOS and android bundles and links to docs


### Screenshots

#### updated modal:
![image](https://github.com/user-attachments/assets/0a90831a-847f-4ef3-a018-dd2af7aa5655)

#### empty state:
![image](https://github.com/user-attachments/assets/0d4e04fe-6af2-43ab-a5b5-245bba70f551)

#### empty search state:
![image](https://github.com/user-attachments/assets/d8b6250e-0757-4ea7-ab45-1dcebfa97d6d)

smeubank committed:
Remove the source maps wizard for react-native projects

The source maps wizard is designed for JavaScript projects. it cannot help developers working with react-native

This PR extends the logic to detect if it is a react-native project and rmeove the wizard on that condition.

We should improve the modal to better handle projects but that is another issue.

smeubank committed:
shufflin some text

remove some repetitive text,  add link to RN source maps docs to top. Since it's kinda hidden below.

smeubank committed:
Adding a link to the source maps page

next we should add context to that page, especially for the empty state, which is most likely for users in this flow

priscilawebdev commented on static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx:
```diff
@@ -482,10 +482,25 @@ export function SourceMapsDebuggerModal({
         <p>
           {t(
-            "It looks like the original source code for this stack frame couldn't be determined when this error was captured. To get the original code for this stack frame, Sentry needs source maps to be configured."
+            'It looks like the original source code for this stack frame. To get the original code for this stack frame, Sentry needs source maps to be uploaded.'
```
I believe you removed part of the first sentence by mistake
```suggestion
            'It looks like the original source code for this stack frame couldn't be determined when this error was captured. To get the original code for this stack frame, Sentry needs source maps to be uploaded.'
```

priscilawebdev commented on static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx:
```diff
@@ -482,10 +482,25 @@ export function SourceMapsDebuggerModal({
+        {isReactNativeSDK({
+          sdkName: sourceResolutionResults.sdkName,
+        }) ? (
+          <Fragment>
+            <WizardInstructionParagraph>
+              {tct(
+                "For React Native projects, source maps should be generated and uploaded automatically during the build process. If your source maps aren't showing up, consult our [link:React Native source maps documentation].",
+                {
+                  link: (
+                    <ExternalLinkWithIcon href="https://docs.sentry.io/platforms/react-native/sourcemaps/" />
+                  ),
+                }
+              )}
+            </WizardInstructionParagraph>
+          </Fragment>
```
`Fragment` is only used to wrap multiple children. We have only 1 in this case, so we can remove it 
```suggestion
         <WizardInstructionParagraph>
              {tct(
                "For React Native projects, source maps should be generated and uploaded automatically during the build process. If your source maps aren't showing up, consult our [link:React Native source maps documentation].",
                {
                  link: (
                    <ExternalLinkWithIcon href="https://docs.sentry.io/platforms/react-native/sourcemaps/" />
                  ),
                }
              )}
            </WizardInstructionParagraph>
```

smeubank committed:
Update static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>

smeubank committed:
Update static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>

smeubank committed:
empty state: rn source maps

added empty state and empty search state, which includes links to react-native source maps docs, and example scripts of what the developers need to do to ensure they're uploading source maps for Android and iOS builds

smeubank committed:
update some copy

smeubank committed:
Merge branch 'master' into (feat)no-sourcemaps-wizard-for-rn

smeubank committed:
update the android scripts

priscilawebdev commented on static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx:
```diff
@@ -1146,15 +1162,32 @@ function UploadedSourceFileWithCorrectDebugIdChecklistItem({
-              link: <ExternalLinkWithIcon href={sourceMapsDocLinks.sourcemaps} />,
-            }
-          )}
+          {tct('You did not upload any artifacts with debug IDs yet.', {
+            link: <ExternalLinkWithIcon href={sourceMapsDocLinks.sourcemaps} />,
+          })}
```
```suggestion
          {tct('You did not upload any artifacts with [link:Debug IDs] yet.', {
            link: <ExternalLinkWithIcon href={sourceMapsDocLinks.debugIds} />,
          })}
```

priscilawebdev replied:
`sourceMapsDocLinks.debugIds` can be `undefined` and we have to check that

priscilawebdev commented on static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx:
```diff
@@ -1213,13 +1259,31 @@ function UploadedSourceMapWithCorrectDebugIdChecklistItem({
-              link: <ExternalLinkWithIcon href={sourceMapsDocLinks.sourcemaps} />,
-            }
-          )}
+          {tct('You did not upload any artifacts with debug IDs yet.', {
+            link: <ExternalLinkWithIcon href={sourceMapsDocLinks.sourcemaps} />,
+          })}
```
same here. Also if we are going to repeat this multiple times, maybe it is worth creating a reusable component

priscilawebdev commented on static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx:
```diff
@@ -1213,13 +1259,31 @@ function UploadedSourceMapWithCorrectDebugIdChecklistItem({
+        {isReactNativeSDK({sdkName: sourceResolutionResults.sdkName}) && (
+          <p>
+            {tct(
+              'Check your [link:source maps uploads] for the missing source maps for debug ID: [debugId]. You should have source maps for both Android and iOS builds.',
```
should we always have Debug IDs capitalized?

priscilawebdev commented on static/app/views/settings/projectSourceMaps/sourceMapsList.tsx:
```diff
@@ -165,6 +170,132 @@ export function SourceMapsList({location, router, project}: Props) {
+            )}
+          </p>
+          <p>
+            {tct(
+              'Be sure during your build you are running these scripts to upload source maps for both iOS and Android:',
+              {}
+            )}
```
```suggestion
            {t(
              'Be sure during your build you are running these scripts to upload source maps for both iOS and Android:'
            )}
```

priscilawebdev commented on static/app/views/settings/projectSourceMaps/sourceMapsList.tsx:
```diff
@@ -165,6 +170,132 @@ export function SourceMapsList({location, router, project}: Props) {
+      {isReactNative && (
+        <Fragment>
+          <p>
+            {tct(
+              'If you are searching for React Native source maps to match specific debugIds, be sure during your build you are running these scripts to upload source maps for both iOS and Android:',
+              {}
+            )}
```
```suggestion
            {t(
              'If you are searching for React Native source maps to match specific Debug IDs, be sure during your build you are running these scripts to upload source maps for both iOS and Android:'
            )}
```

priscilawebdev commented:
Should we have a panel wrapping the empty state? 🤔

![image](https://github.com/user-attachments/assets/3af22335-6e68-4b3a-a535-c1a04d70ebe3)

smeubank committed:
Update static/app/views/settings/projectSourceMaps/sourceMapsList.tsx

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>

smeubank committed:
Update static/app/views/settings/projectSourceMaps/sourceMapsList.tsx

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>

smeubank committed:
Update static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>

krystofwoldrich commented:
This is a great improvement of the UX when source maps are missing for RN project,

what is blocking the merge?

smeubank commented:
oh i didn't realize those changes were shipped and correct 😅 

i can't confirm the accuracy of the RN scrips but i will close
